pallets/sto/src/lib.rs

```diff
@@ -471,48 +524,48 @@ impl<T: Trait> Module<T> {
         fundraiser_id: u64,
         frozen: bool,
     ) -> DispatchResult {
-        Self::ensure_perms_pia(origin, &offering_asset)?;
+        let did = Self::ensure_perms_pia(origin, &offering_asset)?.0;
+        let mut fundraiser = <Fundraisers<T>>::get(offering_asset, fundraiser_id)
+            .ok_or(Error::<T>::FundraiserNotFound)?;
         ensure!(
-            <Fundraisers<T>>::contains_key(offering_asset, fundraiser_id),
-            Error::<T>::FundraiserNotFound
+            fundraiser.status != FundraiserStatus::Closed,
+            Error::<T>::FundraiserClosed
         );
-        <Fundraisers<T>>::mutate(offering_asset, fundraiser_id, |fundraiser| {
-            if let Some(fundraiser) = fundraiser {
-                fundraiser.frozen = frozen
-            }
-        });
+        if frozen {
+            fundraiser.status = FundraiserStatus::Frozen;
+            Self::deposit_event(RawEvent::FundraiserFrozen(did, fundraiser_id));
+        } else {
+            fundraiser.status = FundraiserStatus::Live;
+            Self::deposit_event(RawEvent::FundraiserUnfrozen(did, fundraiser_id));
```

Comment:
I'd prefer one event where `frozen` is simply embedded as a flag; makes the code simpler.

Reply:
Since the extrinsics are different for freezing and unfreezing, I'd prefer to keep the events distinct as well. Sure, it requires a couple more LOC but there's no performance impact.

Reply:
Well actually there's a branch, but the impact is minor... But I would have preferred one extrinsic also, but let's refactor at least...

Reply:
We would've needed a branch anyway because the status needs updating. Removing the branch at one level up would've required code duplication.

I really don't see the need for refactoring here. It will only increase the LOC and add a redirection.